src/nemos/basis.py

```diff
@@ -1511,18 +1670,22 @@ def _check_width(width: float) -> None:
     @support_pynapple(conv_type="numpy")
     @check_transform_input
     @check_one_dimensional
-    def __call__(self, sample_pts: ArrayLike, rescale_samples=True) -> FeatureMatrix:
+    def __call__(
+        self,
+        sample_pts: ArrayLike,
+        rescale_samples=True,
+    ) -> FeatureMatrix:
         """Generate basis functions with given samples.
 
         Parameters
         ----------
         sample_pts :
             Spacing for basis functions, holding elements on interval [0, 1], Shape (number of samples, ).
-
-        Returns
-        -------
-        basis_funcs :
-            Raised cosine basis functions, shape (n_samples, n_basis_funcs).
+        rescale_samples :
```

Comment:
removed, and added a  private flag at init that define if the RaisedCosineLinear should rescale or not (if called by the subclass it should not, otherwise it always have to)